File: java/src/main/java/com/purescraps/purehtml/transformers/types/JSONTransformer.java
package com.purescraps.purehtml.transformers.types;

import java.util.List;

import org.yaml.snakeyaml.Yaml;

import com.purescraps.purehtml.backend.PureHTMLNode;
import com.purescraps.purehtml.transformers.TransformParams;
import com.purescraps.purehtml.transformers.Transformer;

public class JSONTransformer extends Transformer {
    @SuppressWarnings("unused")
    private List<String> args;

    @SuppressWarnings("unused")
    public JSONTransformer(List<String> args) {
        this.args = args;
    }

    @SuppressWarnings("unused")
    public JSONTransformer() {
        this.args = null;
    }

    public void setArgs(List<String> args) {
        this.args = args;
    }

    // Static method to return the name of the transformer
    public String getName() {
        return "json";
    }

    // The transform method that parses the element's text content as JSON
    @Override
    public Object transform(TransformParams params) {
        PureHTMLNode element = params.getElement();

        if (element == null) {
            return null;
        }

        // `<script>`/`<style>` content is not exposed through text(), so prefer the
        // raw node data (falls back to text() for regular elements).
        String content = element.data();
        if (content == null || content.isBlank()) {
            content = element.text();
        }

        // JSON is a subset of YAML, so parsing it with the YAML parser yields plain
        // Map/List/String/Number/Boolean values, same as every other config value.
        return new Yaml().load(content);
    }
}

File: python/purehtml/transformers/types/json_transformer.py
import json
from abc import ABC
from typing import List

from purehtml.transformers.transform_params import TransformParams
from purehtml.transformers.transformer import Transformer


class JSONTransformer(Transformer, ABC):
    def __init__(self, args: List[str] = None):
        """
        Initialize the transformer with optional arguments.
        :param args: Arguments to configure the transformer (unused in this case).
        """
        super().__init__()
        self.args = args if args is not None else []

    def set_args(self, args: List[str]):
        """
        Set the list of attributes to retrieve.
        :param args: A list of attribute names.
        """
        self.args = args

    def get_name(self) -> str:
        """
        Return the name of the transformer.
        :return: The name of the transformer.
        """
        return "json"

    def transform(self, params: TransformParams) -> object:
        """
        Parse the text content of the element as JSON.
        :param params: The parameters that contain the element to extract the JSON from.
        :return: The parsed JSON value, or None if the element does not exist.
        """
        element = params.get_element()

        if element is None:
            return None

        return json.loads(element.text())
